internal/core/application/trade_service.go

```diff
@@ -540,16 +570,72 @@ func (t *tradeService) getUnspentsBlindingsAndDerivationPathsForAccount(
 	return unspents, utxos, blindingKeysByScript, derivationPaths, nil
 }
 
-func (t *tradeService) startObservingAddressesAndTx(addressesToObserve []domain.AddressInfo, txid string) {
-	t.blockchainListener.StartObserveTx(txid)
+func (t *tradeService) unlockUnspentsForTrade(trade *domain.Trade) {
+	p, _ := pset.NewPsetFromBase64(trade.PsetBase64)
+	keyLen := len(p.Inputs)
+	unspentKeys := make([]domain.UnspentKey, keyLen, keyLen)
+
+	for i, in := range p.UnsignedTx.Inputs {
+		unspentKeys[i] = domain.UnspentKey{
+			TxID: bufferutil.TxIDFromBytes(in.Hash),
+			VOut: in.Index,
+		}
+	}
 
-	for _, info := range addressesToObserve {
-		t.blockchainListener.StartObserveAddress(
-			info.AccountIndex,
-			info.Address,
-			info.BlindingKey,
+	count, err := t.unspentRepository.UnlockUnspents(
+		context.Background(),
+		unspentKeys,
+	)
+	if err != nil {
+		log.Warnf(
+			"unable to unlock unspents for trade with id %s. You must run "+
+				"ReloadUtxo RPC as soon as possible to restore the utxo set of the "+
+				"internal wallet. Error: %v", trade.ID, err,
 		)
-		time.Sleep(200 * time.Millisecond)
+	}
+
+	log.Debugf("unlocked %d unspents", count)
+}
+
+func (t *tradeService) checkTradeExpiration(
+	tradeTxID string,
+	selectedUnspentKeys []domain.UnspentKey,
+) {
+	// admit a tollerance of 1 minute past the expiration time.
+	time.Sleep(time.Duration(t.expiryDuration+60) * time.Second)
+	ctx := context.Background()
+
+	// if the trade is expired it's required to unlock the unspents used as input
+	// and to bring the trade to failed status
```

Comment:
not sure about that: a trade should be expired status, it's not failed per se IMHO

Reply:
Ok I'm going to add a new `ExpiredStatus` at domain level, but we'll need also a new `Expire()` method in charge of updating the status accordingly. Just like we have `Propose()`, `Accept()`, and so on for other statuses.